volatility3/framework/plugins/windows/crashinfo.py

```diff
@@ -0,0 +1,83 @@
+# This file is Copyright 2021 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+import datetime
+from volatility3.framework import interfaces, renderers
+from volatility3.framework.configuration import requirements
+from volatility3.framework.renderers import format_hints, conversion
+from volatility3.framework.objects import utility
+
+vollog = logging.getLogger(__name__)
+
+class Crashinfo(interfaces.plugins.PluginInterface):
+    _required_framework_version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls):
+        return [
+            requirements.TranslationLayerRequirement(name = 'primary',
+                                                     description = 'Memory layer for the kernel',
+                                                     architectures = ["Intel32", "Intel64"]), 
+            ]
+        
+    def _generator(self, layer):
```

Comment:
Typing information would help draw out the implicit assumption here.  Layer needs to be a CrashDump layer here, and that needs to have been checked by whatever's handing in layer.

Reply:
That was fixed in d2db22c